src/index.js

```diff
@@ -487,45 +491,41 @@ class TerserPlugin {
         }
 
         if (cache.isEnabled() && !taskResult.error) {
-          taskResult = await cache.store(task, taskResult).then(
-            () => taskResult,
-            () => taskResult
-          );
+          try {
+            await cache.store(task, taskResult);
+          } catch (_) {} // eslint-disable-line no-empty
         }
 
-        task.callback(taskResult);
+        await task.callback(taskResult);
 
         return taskResult;
       };
 
       scheduledTasks.push(
-        limit(() => {
+        limit(async () => {
           const task = getTaskForAsset(assetName).next().value;
 
           if (!task) {
             // Something went wrong, for example the `cacheKeys` option throw an error
-            return Promise.resolve();
+            return null;
           }
 
           if (cache.isEnabled()) {
-            return cache.get(task).then(
-              (taskResult) => task.callback(taskResult),
-              () => enqueue(task)
-            );
+            try {
+              const taskResult = await cache.get(task);
+              return task.callback(taskResult);
+            } catch (_) {} // eslint-disable-line no-empty
           }
 
           return enqueue(task);
         })
       );
     }
 
-    return Promise.all(scheduledTasks).then(() => {
-      if (worker) {
-        return worker.end();
-      }
-
-      return Promise.resolve();
-    });
+    await Promise.all(scheduledTasks);
+    if (worker) {
+      await worker.end();
+    }
```

Comment:
But feel free to send a PR with refactor on `await`/`async`

Reply:
can i just push the changes to the same pr and remove the source-map changes?

Reply:
Let's do it in separate PR